snippets/custom-elements-use.md

```diff
@@ -0,0 +1,138 @@
+---
+title: custom elements use
+author: lironHazan
+twitter: lironn_h
+level: advanced
+tags:
+- tip
+- WebComponents
+---
+
+# Content
+It is possible to use Web Components in your Angular application.
+
+Create a custom element according to the wc standards.
+
+```javascript
+export class MightyToaster extends HTMLElement {
+
+  constructor() {
+      super();
+
+      const shadowRoot = this.attachShadow({mode: 'open'});
+
+      shadowRoot.innerHTML = `
+            <style>
+               :host {
+                display: none;
+                position: fixed;
+                z-index: 1;
+                top: 1rem;
+                width: 100%;
+                background-color: #FF00FF; 
+                font-family: 'Bowlby One', cursive;
+                text-align: center; 
+                padding: 16px;
+                }
+                :host([show]) {
+                  display: block;
+                }
+            </style>
+            <div id="mighty-toast"></div>
+        `;
+
+      this.toast = this.shadowRoot.querySelector('#mighty-toast');
+
+    }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+   if(name === 'pop' && newValue.showAlert === oldValue.showAlert) {
+       this.setAttribute('show', true);
+       setTimeout(() => { 
+       this.removeAttribute('show');
+       }, 3000);
+   }
+  }
+
+  get pop() {
+    return this.getAttribute('pop');
+  }
+
+   set pop(value) {
+    this.setAttribute('pop', value);
+  }
+
+  get text() {
+    return this.getAttribute('text');
+  }
+
+  set text(value) {
+    this.toast.innerHTML = value;
+    this.setAttribute('text', value);
+  }
+
+  get color() {
+    return this.getAttribute('color');
+  }
+
+  set color(value) {
+    this.toast.style.backgroundColor = value;
+    this.setAttribute('color', value);
+  }
+
+  }
+  
+MightyToaster.observedAttributes = [ 'text', 'pop'];
+
+customElements.define('mighty-toast', MightyToaster);
+```
+
+Use it as any other component on your Angular template!
```

Comment:
I think this should be in the Content section and wc code in component code. This will fix snippet validation problem.

Reply:
thanks, I'll try fixing it